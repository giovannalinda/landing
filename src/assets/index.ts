export enum Assets {
  Background = '/bg-header.png',
  About = '/computer.png',
  ImgWork1 = '/imgwork1.png',
  ImgWork2 = '/imgwork2.png',
  ImgWork3 = '/imgwork3.png',
  ImgWork4 = '/imgwork4.png',
  ImgWork5 = '/imgwork5.png',
  ImgWork6 = '/imgwork6.png',
  Dot = '/dot.svg',
  LogoYoutube = '/youtubelogo.svg',
  LogoSketch = '/sketchlogo.svg',
  LogoGoogle = '/googlelogo.svg',
  LogoAdobe = '/adobelogo.svg',
  Logo = '/logo.svg',
  Quote = '/quote.svg',
  Shapes = '/shapes.svg',
}
