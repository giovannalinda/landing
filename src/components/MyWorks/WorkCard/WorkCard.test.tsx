import { render, screen, userEvent } from '~/__helpers__/app-tests'

import Image from 'next/image'

import { Assets } from '~/assets'
import { WorkCard } from '.'

describe('WorkCard', () => {
  it('should render correctly', async () => {
    render(
      <WorkCard href='https://url.com' title='Eduguard - E-learning Website'>
        <Image src={Assets.ImgWork1} width='424' height='332' />
      </WorkCard>,
    )

    expect(
      await screen.findByText(/Eduguard - E-learning Website/i),
    ).toBeInTheDocument()
    expect(await screen.findByRole('img')).toBeInTheDocument()
  })

  it('should match snapshot', async () => {
    const { container } = render(
      <WorkCard href='https://url.co' title='Eduguard - E-learning Website'>
        <Image src={Assets.ImgWork1} width='424' height='332' />
      </WorkCard>,
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it("should render 'view project' when user hover button", async () => {
    render(
      <WorkCard href='https://url.co' title='Eduguard - E-learning Website'>
        <Image src={Assets.ImgWork1} width='424' height='332' />
      </WorkCard>,
    )

    userEvent.hover(
      await screen.findByRole('button', {
        name: /Eduguard - E-learning Website/i,
      }),
    )

    expect(await screen.findByText(/View project/i)).toBeInTheDocument()
  })
})
