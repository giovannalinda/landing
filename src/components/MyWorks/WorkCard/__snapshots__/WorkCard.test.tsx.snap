// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkCard should match snapshot 1`] = `
<button
  aria-label="Eduguard - E-learning Website"
  class="sc-hTtwUo eEkUeT"
>
  <span
    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
  >
    <span
      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
    >
      <img
        alt=""
        aria-hidden="true"
        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27424%27%20height=%27332%27/%3e"
        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
      />
    </span>
    <img
      data-nimg="intrinsic"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
    />
    <noscript />
  </span>
  <div
    class="sc-jgbSNz gboEIk"
  >
    <h3>
      Eduguard - E-learning Website
    </h3>
    <div
      class="sc-gsnTZi freJdh"
    >
      <span>
        VIEW PROJECT
      </span>
      <svg
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M23.0677 11.9929L18.818 7.75739L17.4061 9.17398L19.2415 11.0032L0.932469 11.0012L0.932251 13.0012L19.2369 13.0032L17.4155 14.8308L18.8321 16.2426L23.0677 11.9929Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</button>
`;
